vllm/v1/core/sched/scheduler.py

```diff
@@ -1394,7 +1493,11 @@ def _free_blocks(self, request: Request):
         del self.requests[request.request_id]
 
     def get_num_unfinished_requests(self) -> int:
-        return len(self.waiting) + len(self.running)
+        num_waiting = sum(
+            req.status != RequestStatus.WAITING_FOR_STREAMING_REQ
+            for req in self.waiting
+        )
```

Comment:
For performance reasons it might be better to maintain a separate count for these. This could become expensive if the waiting queue is very long (even when folks aren't using input streaming)